api/src/main/java/org/pmiops/workbench/api/AppsController.java

```diff
@@ -78,6 +76,37 @@ public ResponseEntity<EmptyResponse> updateApp(
 
   @Override
   public ResponseEntity<ListAppsResponse> listAppsInWorkspace(String workspaceNamespace) {
-    throw new UnsupportedOperationException("API not supported.");
+    DbWorkspace dbWorkspace = workspaceService.lookupWorkspaceByNamespace(workspaceNamespace);
+    validateCanPerformApiAction(dbWorkspace);
+
+    ListAppsResponse response = new ListAppsResponse();
+    response.addAll(leonardoApiClient.listAppsInProject(dbWorkspace.getGoogleProject()));
+    return ResponseEntity.ok(response);
+  }
+
+  /**
+   * Validates user is allowed to perform acc action.
+   *
+   * <p>App ACTION requires:
+   *
+   * <ul>
+   *   <li>Feature is enabled
+   *   <li>User compute is not suspend due to security reason (e.g. egress alert)
+   *   <li>User is OWNER or WRITER of the workspace
+   *   <li>Workspace has valid billing.
```

Comment:
Actually I don't know the historical reason of having this because Google will reject any modifications on a project if billing is invalid. 
I copied this from RuntimeController, reading the comment again, i think we can just having it for Create/Update/Localize